### ISSUE TITLE
`clippy::collapsible_match` code suggestion is not valid Rust

### ISSUE THREAD

**Issue by ChaiTRex** [rust-lang/rust-clippy#16894]
### Summary

`clippy::collapsible_match` lint fails to give a valid suggested code change when two arms are on the same line. For example, `12 => if a { b } else { c }` should say `12 if a => b, 12 => c`, but it says `12f a => { b }`.

### Reproducer

Code ([playground](https://play.rust-lang.org/?edition=2024&gist=1006beb9ab02fa6bbe8cb7d8c2d151c7)):

```rust
pub fn f() {
    let a = 5_u8;
    let b = a;
    _ = match a {
        11 => 0, 12 => if a == b { 1 } else { 0 },
        13 => if a > b { 1 } else { 0 }, _ => 0,
    };
}
```

Current output:

```
    Checking playground v0.0.1 (/playground)
warning: this `if` can be collapsed into the outer `match`
 --> src/lib.rs:5:24
  |
5 |         11 => 0, 12 => if a == b { 1 } else { 0 },
  |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.95.0/index.html#collapsible_match
  = note: `#[warn(clippy::collapsible_match)]` on by default
help: collapse nested if block
  |
5 -         11 => 0, 12 => if a == b { 1 } else { 0 },
5 +         11 => 0, 12f a == b => { 1 },
  |

warning: this `if` can be collapsed into the outer `match`
 --> src/lib.rs:6:15
  |
6 |         13 => if a > b { 1 } else { 0 }, _ => 0,
  |               ^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.95.0/index.html#collapsible_match
help: collapse nested if block
  |
6 -         13 => if a > b { 1 } else { 0 }, _ => 0,
6 +         13f a > b => { 1 }, _ => 0,
  |

warning: `playground` (lib) generated 2 warnings (run `cargo clippy --fix --lib -p playground -- ` to apply 2 suggestions)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.65s
```

Desired output:

```
    Checking playground v0.0.1 (/playground)
warning: this `if` can be collapsed into the outer `match`
 --> src/lib.rs:5:24
  |
5 |         11 => 0, 12 => if a == b { 1 } else { 0 },
  |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.95.0/index.html#collapsible_match
  = note: `#[warn(clippy::collapsible_match)]` on by default
help: collapse nested if block
  |
5 -         11 => 0, 12 => if a == b { 1 } else { 0 },
5 +         11 => 0, 12 if a == b => 1, 12 => 0,
  |

warning: this `if` can be collapsed into the outer `match`
 --> src/lib.rs:6:15
  |
6 |         13 => if a > b { 1 } else { 0 }, _ => 0,
  |               ^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.95.0/index.html#collapsible_match
help: collapse nested if block
  |
6 -         13 => if a > b { 1 } else { 0 }, _ => 0,
6 +         13 if a > b => 1, 13 => 0, _ => 0,
  |

warning: `playground` (lib) generated 2 warnings (run `cargo clippy --fix --lib -p playground -- ` to apply 2 suggestions)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.65s
```

### Version

```text
rustc 1.97.0-nightly (66da6cae1 2026-04-20)
binary: rustc
commit-hash: 66da6cae1a6f12e9585493ab8f8f19cf753091fd
commit-date: 2026-04-20
host: x86_64-unknown-linux-gnu
release: 1.97.0-nightly
LLVM version: 22.1.2
```

### Additional Labels

`I-suggestion-causes-error`

**Comment by profetia** [comment]
Should be fixed in #16749

**Comment by QEEK-AI** [comment]
**QEEK.AI Agent:** 

"Thanks for the clear report and reproducer! This is definitely a bug in the suggestion generation the output `12f a == b` is completely invalid Rust syntax".

**What's happening:**

The lint is trying to collapse:
```rust
12 => if a == b { 1 } else { 0 }
```

Into:
```rust
12 if a == b => 1, 12 => 0
```

But the suggestion generation seems to be concatenating the pattern and condition incorrectly, producing `12f` instead of `12 if`. This looks like a string formatting bug where the space between the pattern and `if` is being omitted, and the arrow `=>` is also malformed.

**Root cause suspicion:**

The lint likely extracts the pattern text and the condition text separately, then concatenates them without proper spacing/formatting when multiple match arms appear on the same line. The single-line case probably isn't being handled correctly in the suggestion span calculation.

**Related:**

This connects to #16749 which you already linked. These may be the same underlying issue with span/suggestion handling in `collapsible_match`.

cc @rust-lang/clippy, this is a clear `I-suggestion-causes-error` bug. The suggestion machinery seems to be producing malformed output when the match arm and if condition are on the same line as other arms.

Would you be able to check if this reproduces with the latest nightly? Also, does it work correctly if you reformat the code so each arm is on its own line? That would help narrow down whether this is specifically about multi-arm lines.
